listenbrainz/db/feedback.py

```diff
@@ -45,66 +47,111 @@ def delete(feedback: Feedback):
         )
 
 
-def get_feedback_by_user_id(user_id: int):
-    """ Get a list of recording feedbacks given by the user.
+def get_feedback_for_user(user_id: int, limit: int, offset: int, score: int = None):
+    """ Get a list of recording feedbacks given by the user
+
         Args:
             user_id: the row ID of the user in the DB
+            score: the score value by which the results are to be filtered. If 1 then returns the loved recordings,
+                   if -1 returns hated recordings.
+            limit: number of rows to be returned
+            offset: number of feedback to skip from the beginning
+
         Returns:
             A list of Feedback objects
     """
 
+    args = {"user_id": user_id, "limit": limit, "offset": offset}
+    query = """ SELECT user_id, "user".musicbrainz_id AS user_name, recording_msid::text, score
+                  FROM recording_feedback
+                 JOIN "user"
+                  ON "user".id = recording_feedback.user_id
+                 WHERE user_id = :user_id """
+
+    if score:
+        query += "AND score = :score"
+        args["score"] = score
+
+    query += """ ORDER BY recording_feedback.created DESC
+                 LIMIT :limit OFFSET :offset """
+
     with db.engine.connect() as connection:
-        result = connection.execute(sqlalchemy.text("""
-            SELECT user_id, recording_msid::text, score
-              FROM recording_feedback
-          ORDER BY created DESC
-            """), {
-                'user_id': user_id
-            }
-        )
+        result = connection.execute(sqlalchemy.text(query), args)
         return [Feedback(**dict(row)) for row in result.fetchall()]
 
 
-def get_feedback_by_user_id_and_score(user_id: int, score: int):
-    """ Get a list of recording feedbacks of particular type (loved or hated) given by the user.
-        If score is +1 then get loved recordings, if -1 then get hated recordings.
+def get_feedback_count_for_user(user_id: int):
+    """ Get total number recording feedbacks given by the user
```

Comment:
spelling: total number of recording feedbacks

Reply:
Oops